### PR TITLE
Fix flaky test by changing assertion method

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

Today I learned about the world's most misleading method name `assertCountEqual` which
> Asserts that two iterables have the same elements, the same number of
        times, without regard to order.
        
which is ideal when order does not matter in a test. This way, if for whatever unknown reason a list's order is not guaranteed, we don't get intermittent failures. 
### Safety story

Test change only.
### Automated test coverage

Only change here is in a test.
### QA Plan

No QA.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by gherceg** [comment]
fyi @millerdev this should fix the failing test on https://github.com/dimagi/commcare-hq/pull/31236

**Review by calellowitz — APPROVED** [review]

File: corehq/apps/enterprise/tests/test_forms.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_update_settings_clears_cache(self, clear_domain_caches):
         domains_cache_cleared = [
             arg[0][0] for arg in clear_domain_caches.call_args_list
         ]
-        self.assertListEqual(
+        self.assertCountEqual(
```
Comment (millerdev) [inline corehq/apps/enterprise/tests/test_forms.py:114]: I agree, `assertCountEqual` is a confusing name. Could also do the same with a set comparison.
```py
        domains_cache_cleared = {
            arg[0][0] for arg in clear_domain_caches.call_args_list
        }
        self.assertEqual(
            domains_cache_cleared,
            {'test-emws-form-001', 'test-emws-form-002'},
        )
```

**Review by millerdev — COMMENTED** [review]
Thanks for fixing.

File: corehq/apps/enterprise/tests/test_forms.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/enterprise/tests/test_forms.py:114]: Yeah that is how I've been addressing these, but felt like there must be some method in existence that handles lists directly. Despite the confusing name, I think I still prefer using it as it is very explicit about order not mattering _and_ catches unexpected duplicates if that is applicable.

Comment (millerdev) [reply corehq/apps/enterprise/tests/test_forms.py:114]: I think it's a lot less efficient than set comparison, but in this case that does not matter.

**Pull request merged**